cps/tasks/metadata_extract.py

```diff
@@ -145,9 +153,11 @@ def run(self, worker_thread):
             self._remove_shorts_from_db(conn)
             requested_urls = self._fetch_requested_urls(conn)
             if not requested_urls:
-                return
+                self.message = f"{self.media_url_link} failed: Video not available."
+                conn.execute("DELETE FROM media WHERE path = ?", (self.media_url,))
+                self.stat = STAT_FAIL
```

Comment:
This gets overridden with STAT_FINISH_SUCCESS at line 178. Is it ok?

Reply:
Also, previously this line was skipped when requested_urls is empty, now it's executed every time:

```python
self._add_download_tasks_to_worker(requested_urls)
```

Are there any unwanted side-effects?

Reply:
Yes. STAT_FINISH_SUCCESS and STAT_FAIL both end the task.

Reply:
> Also, previously this line was skipped when requested_urls is empty, now it's executed every time:
> 
> ```python
> self._add_download_tasks_to_worker(requested_urls)
> ```
> 
> Are there any unwanted side-effects?

This one send every requested_url to the download task. It could be done in a single task, but was discarded in favor of individual task for every download.

Reply:
The list of requested URLs is retrieved from xklb-metadata.db. It's basically a search for every row which have no text in its error column and path is a URL. 

This list can be empty if these conditions are not met.

Reply:
> The list of requested URLs is retrieved from xklb-metadata.db. It's basically a search for every row which have no text in its error column and path is a URL.
> 
> This list can be empty if these conditions are not met.

My original question was: is it ok to call _add_download_tasks_to_worker() even when requested_urls is an empty list?

I looked at the body of _add_download_tasks_to_worker(), and it does nothing when called with an empty list.

So... OK.